lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -331,6 +331,7 @@ impl ShardReplicaSet {
             clock_set: Default::default(),
         };
 
+        // TODO: Should we consider `ReshardingScaleDown` state, when counting active replicas?
```

Comment:
With updates in general, I'd say:
- always require at least 1 active replica to accept it (or none if there are no active replicas)
- always require all resharding replicas to accept all operations

Later, if we also implement recovery of replicas that are in resharding down state, we may reduce the requirement for number of resharding replicas to accept it to 1 as well.

Basically, we'd handle active and resharding replicas separately in terms of counting them. If we get an error on either, simply return the error back to the user.

If we change the write consistency factor the requirement of 1 would be larger if there are enough replicas.

Or did you have some other mechanism in mind?